### PR TITLE
Parameterize cuckoo filter on hash family.

### PR DESCRIPTION
**Pull request by jbapple-cloudera**
Add two hash families: two-independent multiply-shift and simple
tabulation. The interface for the hash family is like that used in
std::unordered_*.

### CODE REVIEW

**Comment by jbapple-cloudera** [comment]
@dave-andersen This is now rebased on HEAD and ready for review.

File: src/hashutil.h [hunk 1]
```diff
@@ -40,22 +40,54 @@ class HashUtil {
...
+    for (unsigned i = 0; i < sizeof(key); ++i) {
+      result ^= tables_[i][reinterpret_cast<uint8_t *>(&key)[i]];
+    }
+    return result;
+  }
+};
+}
 
-#endif  // CUCKOO_FILTER_HASHUTIL_H_
+#endif  // #ifndef _HASHUTIL_H_
```
Comment (apc999) [inline src/hashutil.h]: emm, here `// CUCKOO_FILTER_HASHUTIL_H_`  is Google C++ style convention. Any reason to change?

File: src/cuckoofilter.h [hunk 2]
```diff
@@ -214,16 +217,11 @@ Status CuckooFilter<ItemType, bits_per_item, TableType>::Delete(
   return Ok;
 }
 
-template <typename ItemType, size_t bits_per_item,
-          template <size_t> class TableType>
-std::string CuckooFilter<ItemType, bits_per_item, TableType>::Info() const {
+template <typename ItemType, size_t bits_per_item, template <size_t> class TableType,
+    typename HashFamily>
```
Comment (apc999) [inline src/cuckoofilter.h]: is indentation correct here?

File: src/hashutil.h (hunk 1 above)
Comment (jbapple-cloudera) [inline src/hashutil.h]: Don't remember why I did this; undone.

File: src/cuckoofilter.h (hunk 2 above)
Comment (jbapple-cloudera) [inline src/cuckoofilter.h]: Changed to match the file

**Comment by jbapple-cloudera** [comment]
Fixed your comments - thanks!

**Pull request merged**